test/src/hidden_ranges/hidden_line_ranges_test.dart

```diff
@@ -55,6 +55,18 @@ void main() {
         expect(midBreakpointsRanges.cutLineIndexIfVisible(99), null);
         expect(midBreakpointsRanges.cutLineIndexIfVisible(100), 12);
         expect(midBreakpointsRanges.cutLineIndexIfVisible(101), 13);
+
+        expect(midBreakpointsRanges.recoverLineIndex(0), 0);
+        expect(midBreakpointsRanges.recoverLineIndex(1), 1);
+        expect(midBreakpointsRanges.recoverLineIndex(2), 4);
+        expect(midBreakpointsRanges.recoverLineIndex(3), 5);
+        expect(midBreakpointsRanges.recoverLineIndex(4), 6);
+        expect(midBreakpointsRanges.recoverLineIndex(5), 9);
+        expect(midBreakpointsRanges.recoverLineIndex(6), 10);
+        expect(midBreakpointsRanges.recoverLineIndex(7), 11);
+        expect(midBreakpointsRanges.recoverLineIndex(11), 15);
+        expect(midBreakpointsRanges.recoverLineIndex(12), 100);
+        expect(midBreakpointsRanges.recoverLineIndex(13), 101);
       });
     });
```

Comment:
Now that out-of-range causes errors, we need a test for empty text that ensures `0 <-> 0`.